pandas/core/indexes/datetimes.py

```diff
@@ -223,13 +239,13 @@ def __new__(cls, data=None,
             verify_integrity = True
 
         if data is None:
+            result = cls._generate_range(start, end, periods,
```

Comment:
Previously, if you passed a bad set of arguments (e.g. forgot `freq`), then we would warn and then raise an exception.